Homework 12

Dragos-Andrei-75 opened the pull request:
Modified classes:
- UserService
    - defined resendVerificationCode method which checks if the verification code sent to the user is about to expire and, if so, sends him another verification code;
- LibrarianService
    - defined resendVerificationCode method which checks if the verification code sent to the user is about to expire and, if so, sends him another verification code;
- UserController
    - defined resendVerificationCode endpoint;
- LibrarianController
    - defined resendVerificationCode endpoint;

Dragos-Andrei-75 committed:
Homework 12

Modified classes:
- UserService
    - defined resendVerificationCode
      method which checks if the
      verification code sent to the user is
      about to expire and, if so, sends
      him another verification code;
- LibrarianService
    - defined resendVerificationCode
      method which checks if the
      verification code sent to the user is
      about to expire and, if so, sends
      him another verification code;
- UserController
    - defined resendVerificationCode
      endpoint;
- LibrarianController
    - defined resendVerificationCode
      endpoint;

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/service/LibrarianService.java:
```diff
@@ -99,13 +99,22 @@ public void sendVerificationCode(Librarian librarian) {
     }
 
+    public void resendVerificationCode(Librarian librarian) {
+        if (librarian.getVerificationCodeGenerationTime() == null) throw new RuntimeException("No verification code previously generated for librarian.");
```
acolade dupa if, pls - stiu ca merge si fara, dar nu se mai prea practica stilul

marcopacurariu3 replied:
cred ca e mai degraba bad request asta, sincer, pentru ca nu e un bug la noi, e de la client gresit ca a cerut asta

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/service/LibrarianService.java:
```diff
@@ -99,13 +99,22 @@ public void sendVerificationCode(Librarian librarian) {
+
+        Duration elapsedTime = Duration.between(librarian.getVerificationCodeGenerationTime(), LocalDateTime.now());
+
+        if (elapsedTime.toMinutes() < emailService.getVerificationTime() - 1) throw new RuntimeException("Verification code still valid.");
```
nu conteaza, tot il trimiti. dar trebuie sa faci unul nou daca expira in sub 1 minut

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/service/ReservationService.java:
```diff
@@ -71,6 +73,7 @@ public Page<Reservation> findReservationsFromUserByStatus(Long userID, Reservati
         return userRepository.findReservationFromUserByStatus(userID, null, pageable);
     }
 
+    //@Transactional
```
ar trebui sa fie transactional

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/service/UserService.java:
```diff
@@ -93,13 +93,23 @@ public void sendVerificationCode(User user) {
+
+        Duration elapsedTime = Duration.between(user.getVerificationCodeGenerationTime(), LocalDateTime.now());
+
+        if (elapsedTime.toMinutes() < emailService.getVerificationTime() - 1) throw new RuntimeException("Verification code still valid.");
```
analog

Dragos-Andrei-75 committed:
Addressed All Issues from the Previous Review

Addressed all issues from the previous review made by Marco Păcurariu.

Added the Bookstore.postman_collection.json file into the resources folder.

Bookstore.postman_collection.json contains request for testing each endpoint that exist within the project.
